### PR TITLE
VF-20: implement authorization code flow for OIDC

### PR DESCRIPTION
**Pull request by asterns**
To test, need to set up test auth0 application and secret, and set appropriate environment variables.

### CODE REVIEW

**Comment by asterns** [comment]
The diffs are easier to look at if you ignore whitespace (https://github.blog/2018-05-01-ignore-white-space-in-code-review/)

**Comment by sjforman** [comment]
Awesome! I'll review this afternoon.

File: src/Auth.js [hunk 1]
```diff
@@ -20,49 +21,66 @@ export default class Auth {
...
+          });
+        } else {
+          history.replace('/dashboard');
+        }
+      });
+    }
+  }
+  handleAuthentication( {location} ) {
+    let params = queryString.parse(location.search);
+    console.log({params});
```
Comment (sjforman) [inline src/Auth.js]: Do we want to do this in production or is this a stray console.log from development process?

Comment (asterns) [reply src/Auth.js]: stray console.log - just fixed.

File: README.md [hunk 2]
```diff
@@ -66,12 +66,15 @@ createdb votefwd
...
+To test in development, you'll need to have a client id and secret from auth0.
+
 ```
 REACT_APP_URL=http://localhost:3000
 REACT_APP_API_URL=http://localhost:3001/api
 REACT_APP_DATABASE_URL=postgres://<YOURUSERNAME>:@localhost:5432/votefwd
 REACT_APP_DATABASE_DIALECT=postgres
-REACT_APP_AUTH0_CLIENTID=T0oLi22JFRtHPsx0595AAf8p573bxD4d
+REACT_APP_AUTH0_CLIENTID=<YOUR AUTH0 CLIENT ID>
+AUTH0_CLIENT_SECRET=<YOUR AUTH0 CLIENT SECRET>
```
Comment (sjforman) [inline README.md]: I wound up having to also add a `REACT_APP_AUTH0_ID` to get things working locally, but I'm not 100% sure of its function. Let's discuss live.

File: src/Auth.js [hunk 3]
```diff
@@ -20,49 +21,65 @@ export default class Auth {
...
+    } else {
+      this.exchangeAuthorizationCode({code: params.code, nonce: localStorage.getItem("oauth_nonce")})
+      .then((results) => {
+        localStorage.removeItem("oauth_state");
+        localStorage.removeItem("oauth_nonce");
+        const authResult = results.data;
+        return this.handleAuthResult(authResult);
+      })
+      .catch(err => {
+        console.error('generateFormForVoter error', err);
```
Comment (sjforman) [inline src/Auth.js]: stray copy/paste

Comment (asterns) [reply src/Auth.js]: @sjforman - removed stray copy/paste

**Pull request merged**